### ISSUE TITLE
Not compatible with firebase_core: 0.7.0

### ISSUE THREAD

**Issue by zamahaka** [mwaylabs/flutter-apns#44]
`flutter-apns` has intrinsic dependency on `firebase_messaging^7.0.0`.
And `firebase_messaging^7.0.0` doesn't allow for the app to be compiled with `firebase_core: 0.7.0`.

`Pub get` error log:
```
Because firebase_messaging >=7.0.0 <7.0.3 depends on firebase_core ^0.5.0 and no versions of firebase_messaging match >7.0.3 <8.0.0, firebase_messaging >=7.0.0 <7.0.3-∞ or >7.0.3 <8.0.0 requires firebase_core ^0.5.0.
And because firebase_messaging 7.0.3 depends on firebase_core ^0.5.0+1, firebase_messaging ^7.0.0 requires firebase_core ^0.5.0.
Because no versions of flutter_apns match >1.4.1 <2.0.0 and flutter_apns 1.4.1 depends on firebase_messaging ^7.0.0, flutter_apns ^1.4.1 requires firebase_messaging ^7.0.0.
Thus, flutter_apns ^1.4.1 requires firebase_core ^0.5.0.
So, because mymilkman depends on both firebase_core ^0.7.0 and flutter_apns ^1.4.1, version solving failed.
pub get failed (1; So, because mymilkman depends on both firebase_core ^0.7.0 and flutter_apns ^1.4.1, version solving failed.)
```

**Comment by espadon8004** [comment]
Same issue here

**Comment by andzejsw** [comment]
So far as i look, only option is to use https://github.com/mohammedhasna2015/flutter-apns fork, which one doesn't use firebase_messaging, cause it isn't really needed.

**Comment by andzejsw** [comment]
@szotp is there planned updates?

**Comment by szotp** [comment]
I just merged update to firebase_messaging 9.0.0, I'll test it and probably publish new version today.

The point of this plugin was to handle both apns and firebase using common interface, but I see now it would be easier to maitain if people had to do the firebase part themselves. I think I'll a make second package without dependency on firebase, so people can decide what they want. 

flutter_apns_only: handles only apns
flutter_apns: works as before, depends on flutter_apns_only and firebase_messaging
